test/utils/common.py

```diff
@@ -112,7 +112,8 @@ def tempdir():
 
 @contextlib.contextmanager
 def temppath():
-    path = tempfile.mktemp()
+    _, path = tempfile.mkstemp()
```

Comment:
`mkstemp` has the argument `suffix`, that should make `path += '.tmp''` obsolete.

Reply:
Further, `mkstemp` **creates** a file, and then we come up with a different name and create and delete that one, but not the original one. Not good.

Reply:
maybe we should use `tempfile.mkdtemp` here since we use the path as a directory

Reply:
@EnricoMi I have changed this to mkdtemp as that makes the most sense for these tests. Thanks for the feedback!